src/support/metadata/TypeMetadata.ts

```diff
@@ -1,26 +1,49 @@
+import {
+    PackedBuildable,
+    Buildable,
+    BuildableResolver,
+    BuildableResolverAsync,
+    Typeable,
+} from '../Type';
+import { Entity } from '../../Entity';
+
+function isResolverFunction(type: Typeable): type is BuildableResolver {
+    // If the object's name is empty, we will assume it's an anonymous function that resolves the actual type.
+    return ((type as unknown as Function).name?.length === 0);
```

Comment:
Why do we need to cast `as unknown as Function` when `Typeable` describes the function options?

Reply:
Errr, that was probably left over from earlier efforts. I tried removing it, and yeah, TS doesn't complain. Will delete that part.